pyart/retrieve/qvp.py

```diff
@@ -111,3 +141,1681 @@ def quasi_vertical_profile(
+        If another type of interpolation is selected masked values will be
+        eliminated from the data points before the interpolation
+    qvp : QVP object or None
+        If it is not None this is the QVP object where to store the data from
+        the current time step. Otherwise a new QVP object will be created
+
+
+    Returns
+    -------
+    qvp : qvp object
+        The computed vertical profile
+
+    """
+    radar_aux = deepcopy(radar)
+    # transform radar into ppi over the required elevation
+    if radar_aux.scan_type == "rhi":
+        target_elevations, el_tol = get_target_elevations(radar_aux)
+        radar_ppi = cross_section_rhi(radar_aux, target_elevations, el_tol=el_tol)
+    elif radar_aux.scan_type == "ppi":
+        radar_ppi = radar_aux
+    else:
+        warn("Error: unsupported scan type.")
+        return None
+
+    if qvp is None:
+        qvp = _create_qvp_object(
+            radar_ppi,
+            field_names,
+            qvp_type="vp",
+            start_time=ref_time,
+            hmax=hmax,
+            hres=hres,
+        )
+
+    # modify metadata
+    if ref_time is None:
+        ref_time = datetime_from_radar(radar_ppi)
+    qvp = _update_qvp_metadata(qvp, ref_time, lon, lat, elev=90.0)
+
+    height = dict()
+    values = dict()
+    for field_name in field_names:
+        height.update({field_name: np.array([], dtype=float)})
+        values.update({field_name: np.ma.array([], dtype=float)})
+
+    for sweep in range(radar_ppi.nsweeps):
+        radar_aux = deepcopy(radar_ppi.extract_sweeps([sweep]))
+
+        # find nearest gate to lat lon point
+        ind_ray, ind_rng, _, _ = find_nearest_gate(
+            radar_aux, lat, lon, latlon_tol=latlon_tol
+        )
+
+        if ind_ray is None:
+            continue
+
+        for field_name in field_names:
+            if field_name not in radar_aux.fields:
+                warn("Field " + field_name + " not in radar object")
+                continue
+
+            height[field_name] = np.append(
+                height[field_name], radar_aux.gate_altitude["data"][ind_ray, ind_rng]
+            )
+            values[field_name] = np.ma.append(
+                values[field_name],
+                radar_aux.fields[field_name]["data"][ind_ray, ind_rng],
+            )
+
+    for field_name in field_names:
+        # Project to vertical grid:
+        qvp_data = project_to_vertical(
+            values[field_name],
+            height[field_name],
+            qvp.range["data"],
+            interp_kind=interp_kind,
+        )
+
+        # Put data in radar object
+        if np.size(qvp.fields[field_name]["data"]) == 0:
+            qvp.fields[field_name]["data"] = qvp_data.reshape(1, qvp.ngates)
+        else:
+            qvp.fields[field_name]["data"] = np.ma.concatenate(
+                (qvp.fields[field_name]["data"], qvp_data.reshape(1, qvp.ngates))
+            )
+
+    return qvp
+
+
+def compute_ts_along_coord(
```

Comment:
Would it be possible to test the non qvp functions as well? I think we are ok on the private functions.